apps/api/src/app/widgets/usecases/get-subscriber-preference/get-subscriber-preference.usecase.ts

```diff
@@ -0,0 +1,62 @@
+import { Injectable } from '@nestjs/common';
+import { SubscriberPreferenceRepository, NotificationTemplateRepository, NotificationTemplateEntity } from '@novu/dal';
+import { GetSubscriberPreferenceCommand } from './get-subscriber-preference.command';
+
+@Injectable()
+export class GetSubscriberPreference {
+  private notificationTemplateRepository = new NotificationTemplateRepository();
+
+  constructor(private subscriberPreferenceRepository: SubscriberPreferenceRepository) {}
+
+  async execute(command: GetSubscriberPreferenceCommand): Promise<any> {
+    const templateList = await this.notificationTemplateRepository.getActiveList(
+      command.organizationId,
+      command.environmentId,
+      true
+    );
+
+    const templatesIds = templateList.map((template) => template._id);
+
+    const subscriberPreferences = await this.subscriberPreferenceRepository.findAll(
+      command.environmentId,
+      command.subscriberId,
+      templatesIds
+    );
+
+    const preferences = templateList.map((template) => {
+      const currSubscriberPreference = subscriberPreferences.find(
+        (preference) => preference._templateId === template._id
+      );
+
+      if (currSubscriberPreference) {
+        return { template: template, preference: currSubscriberPreference };
+      }
+
+      /*
+       * check template default fallback
+       * add here the template(or fallback) is critical as well
+       */
+
+      return getNoSettingFallback(template);
+    });
+
+    /*
+     * check what param the client need
+     * and create response interface
+     * remove useless params on repo query
```

Comment:
Is this a TODO: comment or just comment explaining something? Suggest adding a TODO if this is something you will need to do in the future.

Reply:
Its personal todo for the cycle so nothing more then that.